api/routes/period.js

```diff
@@ -0,0 +1,60 @@
+var express = require("express")
+var router = express.Router();
+var jwt = require('jsonwebtoken');
+var crypto = require('crypto');
+var morgan  = require('morgan');
+const mongoose = require('mongoose');
+const period = require('../models/period')
+
+//Set up default mongoose connection
+module.exports = router;
+
+router.get("/students/update", function(req, res) {
+    period.findByIdAndUpdate(req.body.period_id, req.body.period, function(err, result) {
```

Comment:
One thing is that I implemented a JWT authorization token by storing it in a browser cookie to make sure that the people using our API are logged in users. Before each of these do anything, you can use cookieparser to access the cookies sent in the request and use jwt.verify with the secret key stored in the .env file to check to make sure the user is logged in. Otherwise anybody can change anyones' periods.